privacy/enforcement.go

```diff
@@ -17,14 +17,14 @@ func (e Enforcement) Any() bool {
 }
 
 // Apply cleans personally identifiable information from an OpenRTB bid request.
-func (e Enforcement) Apply(bidRequest *openrtb.BidRequest) {
-	e.apply(bidRequest, NewScrubber())
+func (e Enforcement) Apply(bidRequest *openrtb.BidRequest, isAMP bool) {
+	e.apply(bidRequest, isAMP, NewScrubber())
 }
 
-func (e Enforcement) apply(bidRequest *openrtb.BidRequest, scrubber Scrubber) {
+func (e Enforcement) apply(bidRequest *openrtb.BidRequest, isAMP bool, scrubber Scrubber) {
 	if bidRequest != nil && e.Any() {
-		bidRequest.Device = scrubber.ScrubDevice(bidRequest.Device, e.getIPv6ScrubStrategy(), e.getGeoScrubStrategy())
-		bidRequest.User = scrubber.ScrubUser(bidRequest.User, e.getDemographicScrubStrategy(), e.getGeoScrubStrategy())
+		bidRequest.Device = scrubber.ScrubDevice(bidRequest.Device, !isAMP, e.getIPv6ScrubStrategy(), e.getGeoScrubStrategy())
```

Comment:
Just a nit pick around readability:

The second argument to the `ScrubDevice` function is a flag to indicate if we wanna scrub `mac` and `IFA` values from the device struct and from this code I understand that we don't wanna scrub this for AMP requests. Can we please hold if mac and IFA values be scrubbed or not based on `isAMP` in a different variable and pass that variable? It just helps with readability. Right now for anyone to understand what passing `!isAMP` means in that method call, they need to go and check the function definition. If we create a new variable like `scrubMacAndIFA`, it becomes clear from this line itself. 

Also, other option could be to create something like `ScrubStrategyUser` for device as well.

Reply:
See my previous comment. Device IFA and MAC shouldn't be tied to AMP. We always want to scrub those. There are no issues reported with that logic.